sdk/iot/hub/src/az_iot_hub_client_twin.c

```diff
@@ -251,7 +252,7 @@ AZ_NODISCARD az_result az_iot_hub_client_twin_parse_received_topic(
           &props, az_iot_hub_twin_version_prop, &out_twin_response->version));
 
       out_twin_response->response_type = AZ_IOT_CLIENT_TWIN_RESPONSE_TYPE_DESIRED_PROPERTIES;
-      out_twin_response->request_id = AZ_SPAN_NULL;
+      out_twin_response->request_id = 0;
```

Comment:
Is 0 a sentinel request_id to mean no id?

Reply:
If it was an `int32_t` there is nothing to prevent the user from making it `-1`. Unless we make it so. This would be the only reason to make it `int32_t`. 

In this case, there is no request ID for desired properties. If it is documented well, the user shouldn't be looking at it.